ecs/src/nox/ecs/component/Sprite.cpp

```diff
@@ -0,0 +1,217 @@
+#include <nox/app/graphics/2d/TransformationNode.h>
+#include <nox/ecs/ComponentType.h>
+#include <nox/ecs/EntityManager.h>
+#include <nox/event/IBroadcaster.h>
+#include <nox/logic/graphics/event/SceneNodeEdited.h>
+#include <nox/logic/IContext.h>
+#include <nox/util/json_utils.h>
+
+#include <glm/gtc/matrix_transform.hpp>
+
+void 
+nox::ecs::Sprite::initialize(const Json::Value& componentJsonObject)
+{
+    this->spriteRenderNode = std::make_shared<app::graphics::SpriteRenderNode>();
+
+    this->scale = util::parseJsonVec(componentJsonObject["scale"], glm::vec2(1.0f));
+    this->rotation = componentJsonObject.get("rotation", 0.0f).asFloat();
+    this->flipScale = util::parseJsonVec(componentJsonObject["flipScale"], glm::vec2(1.0f));
+    this->renderingEnabled = true;
+
+    this->entityTransformNode = std::make_shared<app::graphics::TransformationNode>();
+    this->renderTransformNode = std::make_shared<app::graphics::TransformationNode>();
+    this->entityTransformNode->addChild(this->renderTransformNode);
+
+    this->color = util::parseJsonVec(componentJsonObject["color"], glm::vec4(1.0f));
+
+    this->lightMultiplier = componentJsonObject.get("lightMultiplier", 1.0f).asFloat();
+    this->emissiveLight = componentJsonObject.get("emissiveLight", 0.0f).asFloat();
+
+    this->onColorChange(this->color);
+    this->onLightMultiplierChange(this->lightMultiplier);
+    this->onEmissiveLightChange(this->emissiveLight);
+
+    this->renderLevel = componentJsonObject.get("renderLevel", 0).asUInt();
+    this->spriteName = componentJsonObject.get("spriteName", "").asString();
+    this->center = util::parseJsonVec(componentJsonObject["center"], glm::vec2(0.5f));
+    
+    this->spriteRenderNode->setRenderLevel(this->renderLevel);
+    this->spriteRenderNode->addSprite(this->spriteName);
+    this->spriteRenderNode->setActiveSprite(this->spriteName);
+    this->spriteRenderNode->setCenter(this->center);
+
+    onCreate();
+}
+
+void 
+nox::ecs::Sprite::serialize(Json::Value &componentObject)
+{
+    componentObject["scale"] = util::writeJsonVec(this->scale);
+    componentObject["rotation"] = this->rotation;
+    componentObject["flipScale"] = util::writeJsonVec(this->flipScale);
+
+    componentObject["color"] = Json::objectValue;
+    componentObject["color"] = util::writeJsonVec(this->color, util::JsonVecCoordType::COLOR);
+
+    componentObject["spriteName"] = this->spriteName;
+    componentObject["renderLevel"] = this->renderLevel;
+
+    componentObject["center"] = util::writeJsonVec(this->center);
+}
+
+void 
+nox::ecs::Sprite::deactivate()
+{
+    if (this->renderingEnabled == true)
+    {
+        this->broadcastSceneNodeRemoval();
+    }
+}
+
+void 
+nox::ecs::Sprite::activate()
+{
+    if (this->renderingEnabled == true)
+    {
+        this->broadcastSceneNodeCreation();
+    }
+}
+
+void 
+nox::ecs::Sprite::receiveEntityEvent(const ecs::Event& event)
+{
+    if (event.getType() == TypeIdentifier("TransformChange"))
```

Comment:
Don't use String based typeidentifier, use the constants from the EventType header.